statistics/views.py

```diff
@@ -194,6 +258,22 @@ def training_participant(request, rid):
         raise PermissionDenied()
     return render(request, 'statistics/templates/training_participant.html', context)
 
+def studentmaster_ongoing(request, rid):
+    
+    context = {}
+    try:
+        context['model_label'] = 'Workshop / Training'
+        context['model'] = TrainingRequest.objects.get(id=rid)
+        current_batch = TrainingRequest.objects.filter(id=rid)
+        for ab in current_batch:
+            row_data = StudentMaster.objects.filter(batch_id=ab.batch_id)
+        context['collection'] = row_data
+
+    except Exception, e:
```

Comment:
Do not catch all the exceptions, be very specific on which exception you are handling. Here it's `TrainingRequest.DoesNotExist`. Or still better use django's [`get_object_or_404`](https://docs.djangoproject.com/en/2.0/topics/http/shortcuts/#get-object-or-404) function to get the `TrainingRequest` object.